pyro/poutine/replay_messenger.py

```diff
@@ -8,42 +8,17 @@ class ReplayMessenger(Messenger):
     Messenger for replaying from an existing execution trace.
     """
 
-    def __init__(self, trace=None, sites=None):
+    def __init__(self, trace=None, params=None):
         """
         :param trace: a trace whose values should be reused
 
         Constructor.
         Stores trace in an attribute.
         """
         super(ReplayMessenger, self).__init__()
-        assert trace is not None, "must provide trace"
+        assert trace is not None or params is not None, "must provide trace"
```

Comment:
nit: "must provide either trace or params"

Reply:
Actually since this is user-facing we should
```py
if trace is None and params is None:
    raise ValueError("must provide either trace or params")
```